dynamo/vectorfield/vector_calculus.py

```diff
@@ -52,54 +54,56 @@
-        basis: str or None (default: None)
-            The embedding data to use for calculating velocities. If `basis` is either `umap` or `pca`, the
+        init_states: Initial cell states for the historical or future cell state prediction with numerical integration.
+        basis: The embedding data to use for calculating velocities. If `basis` is either `umap` or `pca`, the
             reconstructed trajectory will be projected back to high dimensional space via the `inverse_transform`
             function.
-        vector_field_class: :class:`~scVectorField.vectorfield`
-            If not None, the speed will be computed using this class instead of the vector field stored in adata. You
+        vector_field_class: If not None, the speed will be computed using this class instead of the vector field stored in adata. You
             can set up the class with a known ODE function, useful when the data is generated through simulation.
-        layer: str or None (default: 'X')
-            Which layer of the data will be used for predicting cell fate with the reconstructed vector field function.
-            The layer once provided, will override the `basis` argument and then predicting cell fate in high
+        layer: Which layer of the data will be used for predicting cell fate with the reconstructed vector field function.
+            The layer once provided, will override the `basis` argument and this function will then predict cell fate in high
             dimensional space.
-        dims: int, list, or None (default: None)
-            The dimensions that will be selected for velocity calculation.
-        Qkey: str (default: 'PCs')
-            The key of the PCA loading matrix in `.uns`. Only used when basis is `pca`.
-    Returns
-    -------
-        adata: :class:`~anndata.AnnData`
-            AnnData object that is updated with the `"velocities"` related key in the `.uns`.
+        dims: The dimensions that will be selected for velocity calculation.
+        Qkey: The key of the PCA loading matrix in `.uns`. Only used when basis is `pca`.
+
+    Returns:
+        AnnData object that is updated with the `"velocities"` related key in the `.uns`.
     """
 
     if vector_field_class is None:
-        vf_dict = get_vf_dict(adata, basis=basis)
-        if "method" not in vf_dict.keys():
-            vf_dict["method"] = "sparsevfc"
-        if vf_dict["method"].lower() == "sparsevfc":
-            vector_field_class = SvcVectorField()
-            vector_field_class.from_adata(adata, basis=basis)
-        elif vf_dict["method"].lower() == "dynode":
-            vf_dict["parameters"]["load_model_from_buffer"] = True
-            vector_field_class = vf_dict["dynode_object"]  # dynode_vectorfield(**vf_dict["parameters"])
-        else:
-            raise ValueError("current only support two methods, SparseVFC and dynode")
+        vector_field_class = vector_field_class = get_vf_class(adata, basis=basis)
```

Comment:
Hi, @MukundhMurthy. Is this a typo? Also see line 166.

Reply:
Yes, that was a typo – I had written the `get_vf_class` function to get rid of many duplicated code segments